src/pkg/artifact/manager.go

```diff
@@ -99,24 +96,6 @@ func (m *manager) Create(ctx context.Context, artifact *Artifact) (int64, error)
 	return id, nil
 }
 func (m *manager) Delete(ctx context.Context, id int64) error {
-	// check whether the artifact is referenced by other artifacts
-	references, err := m.dao.ListReferences(ctx, &q.Query{
-		Keywords: map[string]interface{}{
-			"child_id": id,
-		},
-	})
-	if err != nil {
-		return err
-	}
-	if len(references) > 0 {
-		var ids []string
-		for _, reference := range references {
-			ids = append(ids, strconv.FormatInt(reference.ParentID, 10))
-		}
-		return ierror.PreconditionFailedError(nil).
```

Comment:
How do you make sure 412 is returned?

Reply:
This is covered by the foreign key constraint, 412 will returned when got such error